var/spack/repos/builtin/packages/font-util/package.py

```diff
@@ -3,16 +3,102 @@
+         'cab8a44ae329aab7141c7adeef0daf5a'],
+        ['font-bitstream-100dpi', '1.0.3', 'c27bf37e9b8039f93bd90b8131ed37ad'],
+        ['font-bitstream-75dpi', '1.0.3', '4ff6c5d6aebe69371e27b09ad8313d25'],
+        ['font-bitstream-speedo', '1.0.2', 'f0a777b351cf5adefefcf4823e0c1c01'],
+        ['font-bitstream-type1', '1.0.3', 'ff91738c4d3646d7999e00aa9923f2a0'],
+        ['font-cronyx-cyrillic', '1.0.3', '3119ba1bc7f775c162c96e17a912fe30'],
+        ['font-cursor-misc', '1.0.3', 'a0bf70c7e498f1cd8e3fdf6154f2bb00'],
+        ['font-daewoo-misc', '1.0.3', '71a7e2796f045c9d217a19c4e6c25bc1'],
+        ['font-dec-misc', '1.0.3', '5a9242f6b60ecf2b8c5b158322ca2a40'],
+        ['font-ibm-type1', '1.0.3', '2806116e4adcb89d3d5ff5faf65e57c1'],
+        ['font-isas-misc', '1.0.3', 'ecc3b6fbe8f5721ddf5c7fc66f73e76f'],
+        ['font-jis-misc', '1.0.3', 'c48ee5749ae25075d2c7a6111c195e7b'],
+        ['font-micro-misc', '1.0.3', '4de3f0ce500aef85f198c52ace5e66ac'],
+        ['font-misc-cyrillic', '1.0.3', 'e7b13da5325f62dd3f630beade6d2656'],
+        ['font-misc-ethiopic', '1.0.3', '02ddea9338d9d36804ad38f3daadb55a'],
+        ['font-misc-meltho', '1.0.3', '8380696483478449c39b04612f20eea8'],
+        ['font-misc-misc', '1.1.2', '23a79b92275375315129b440206c85b9'],
+        ['font-mutt-misc', '1.0.3', '6c2de53ba514f720e02af48eef28ff32'],
+        ['font-schumacher-misc', '1.1.2', '1f3386a0a690ba8117fc05b501f9f91b'],
+        ['font-screen-cyrillic', '1.0.4', '4cadaf2ba4c4d0f4cb9b4e7b8f0a3019'],
+        ['font-sun-misc', '1.0.3', '87ce97ce0582e76bc4064a4d4d10db09'],
+        ['font-winitzki-cyrillic', '1.0.3',
+         '777c667b080b33793528d5abf3247a48'],
+        ['font-xfree86-type1', '1.0.4', '89c33c5176cd580de6636ad50ce7777b'],
+    ]
+    for f_r in fonts_resource:
+        f = f_r[0]
+        resource(name=f, url=font_baseurl + f + '-' + f_r[1] + '.tar.gz',
+             md5=f_r[2], destination=f, when='fonts=' + f)
+        fonts.append(f)
+
+    variant(
+       'fonts',
+       description='Installs fonts',
+       values=fonts,
+       default=','.join(fonts),
+       multi=True
+    )
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.prepend_path('PATH', self.prefix.bin)
+
+    @run_after('install')
+    def font_install(self):
+        autoconf_args = ['-ifv']
+        p = os.path.join(self.spec['util-macros'].prefix, 'share', 'aclocal')
+        autoconf_args.append('--include={0}'.format(p))
+        p = os.path.join(self.spec.prefix, 'share', 'aclocal')
+        autoconf_args.append('--include={0}'.format(p))
+        fonts = self.spec.variants['fonts'].value
+        autoreconf = which('autoreconf')
+
+        for font in fonts:
+            with working_dir(os.path.join(font, 'spack-src')):
+                autoreconf(*autoconf_args)
+                configure = Executable("./configure")
+                configure('--prefix={0}'.format(self.prefix),
+                          "PKG_CONFIG_PATH=%s/pkgconfig" %
```

Comment:
@tkameyama could you update `PKG_CONFIG_PATH` in `setup_environment`? For example:

```
def setup_environment(...
    ...
    spack_env.prepend_path('PKG_CONFIG_PATH', join_path(self.prefix.lib, 'pkgconfig') 
```

Spack adds other directories to `PKG_CONFIG_PATH` so overriding it here could miss a dependency.

Other than that this looks good.

Reply:
I rewrite it.

Reply:
Wow, that's amazing; and a big step forward!  Have you tried it, does it work?

My one thought now is... do we want to make any of these fonts optional as variants?  Or should they all be installed on all systems?